lib/src/store.dart

```diff
@@ -179,7 +179,7 @@ class Store<State> {
   /// By default, the Stream is async.
   Store(
     this.reducer, {
-    State initialState,
+    State? initialState,
```

Comment:
It looks like requiring the `initialState` to be provided is the best option.

However, it is not clear how to deal with `teardown()` then.

Requiring `initialState` allows to always have a non-null state but how we should clear the state if cannot make it null?

Reply:
`Store.state` never returns `null` from now on. Added `ClosedStoreError` which is thrown when `Store.state` has been accessed after `Store.teardown()` was called.

Reply:
Right now I've been working on https://github.com/fluttercommunity/redux.dart/pull/69 and I was able to make `initialState` a required parameter. Let me know if this doesn't make sense for your use-case.

Reply:
I assume this PR is irrelevant now as #69 is merged.

Reply:
Yes, thanks for the PR, let me know if you have any suggested changes